addons/im_livechat/tests/test_message.py

```diff
@@ -22,6 +23,56 @@ def setUp(self):
             {'name': 'test1', 'login': 'test1', 'email': 'test1@example.com'},
         ])
 
+    @users('emp')
+    def test_chatbot_message_format(self):
+        user = self.env.user
+        channel_info = self.livechat_channel.with_user(user)._open_livechat_mail_channel(
+            anonymous_name='Test Chatbot',
+            previous_operator_id=self.chatbot_script.operator_partner_id.id,
+            chatbot_script=self.chatbot_script,
+            user_id=user.id
+        )
+        mail_channel = self.env['mail.channel'].browse(channel_info['id'])
+        self._post_answer_and_trigger_next_step(
+            mail_channel,
+            self.step_dispatch_buy_software.name,
+            chatbot_script_answer=self.step_dispatch_buy_software
+        )
+        chatbot_message = mail_channel.chatbot_message_ids.mail_message_id[-1:]
+        _format = chatbot_message.message_format()
+        self.assertEqual(_format, [{
+            'id': chatbot_message.id,
+            'body': '<p>Can you give us your email please?</p>',
+            'date': chatbot_message.date,
+            'message_type': 'comment',
+            'subtype_id': (self.env.ref('mail.mt_comment').id, 'Discussions'),
+            'subject': False,
+            'model': 'mail.channel',
+            'res_id': mail_channel.id,
+            'record_name': self.chatbot_script.title,
+            'starred_partner_ids': [],
+            'author': {
+                'id': self.chatbot_script.operator_partner_id.id,
+                'name': self.chatbot_script.title
```

Comment:
Isn't the name deterministic? If so, should be hard-coded instead to ensure proper result.